src/spec_tools/transform.cljc

```diff
@@ -69,6 +70,15 @@
       :else x)
     x))
 
+(defn string->uri [_ x]
+  (if (string? x)
+    (try
+      #?(:clj  (URI/create x)
+         ;; https://stackoverflow.com/questions/11528249/uri-validation-in-javascript
```

Comment:
Closure library provides URI class:

https://google.github.io/closure-library/api/goog.Uri.html

Reply:
I'll change cljs version to use that. Thanks!